src/openagi/agent.py

```diff
@@ -229,6 +229,58 @@ def _force_output(
                 f"LLM did not produce the expected output after {self.max_iterations} iterations."
             )
         return (cont, final_output)
+    
+    def auto_workers_assignment(self,query:str,description:str,task_lists:TaskLists):
+        """
+        Autonomously generates the Workers with the 
+
+        Args:
+            query (str): The query to be processed.
+            description (str): A description of the task.
+            task_lists (TaskLists): The task lists to be processed.
+
+        Returns:
+            str: JSON of the list of Workers that needs to be executed
+        """
+        all_thoughts_and_obs = []
+        output = None
+        previous_task_context = None
+      
+        while not task_lists.all_tasks_completed:
+            iteration = 1
+            max_iterations = self.max_iterations
+
+            cur_task = task_lists.get_next_unprocessed_task()
+            print(cur_task)
+            logging.info(f"**** Executing Task - {cur_task.name} [{cur_task.id}] ****")
+
+            task_to_execute = f"{cur_task.name}. {cur_task.description}"
+            worker_description = f"{cur_task.role} - {cur_task.instructions}"
```

Comment:
@gourab-aiplanet and @suchana-aiplanet 
bere role and instructions are not saved inside Task. Can you look into it.